test/feature-feedback.test.ts

```diff
@@ -142,4 +142,77 @@ describe('FeatureFeedback', () => {
     ) as HTMLLabelElement;
     expect(upvoteButton.classList.contains('error')).to.be.true;
   });
+
+  it('in vote-prompt mode, shows the prompt with vote/comment buttons', async () => {
+    const el = (await fixture(html`
+      <feature-feedback
+        displayMode="vote-prompt"
+        prompt="foobar"
+      ></feature-feedback>
+    `)) as FeatureFeedback;
+
+    const promptContainer = el.shadowRoot!.querySelector('.prompt');
+    const promptText = promptContainer!.querySelector(
+      '.prompt-text'
+    ) as HTMLSpanElement;
+    expect(promptText.innerText).to.equal('foobar');
+
+    const voteButtons = promptContainer!.querySelectorAll('.vote-button');
+    expect(voteButtons.length).to.equal(2);
+
+    const commentButton = promptContainer!.querySelector(
+      '#comment-button'
+    ) as HTMLButtonElement;
+    expect(commentButton).to.exist;
+  });
+
+  it('in vote-prompt mode, submits feedback when clicking a vote button outside the popup', async () => {
+    const service = new MockFeatureFeedbackService();
+    const recaptchaManager = new MockRecaptchaManager();
+
+    const el = (await fixture(html`
+      <feature-feedback
+        displayMode="vote-prompt"
+        featureIdentifier="foo-feature"
+        .featureFeedbackService=${service}
+        .recaptchaManager=${recaptchaManager}
+      ></feature-feedback>
+    `)) as FeatureFeedback;
+
+    const upvoteButton = el.shadowRoot!.querySelector(
+      ':not(#popup) .upvote-button'
+    ) as HTMLLabelElement;
+    upvoteButton.click();
+
+    await el.updateComplete;
+    await nextFrame();
```

Comment:
Ooh I've never used `nextFrame()` before. Does `updateComplete` not wait long enough for the component update?

Reply:
Yeah, I believe this was due to the fact that the component tries to load recaptcha upon voting, before continuing the submit actions. At least when using the test mocks, this induces a one-frame delay beyond the component update. I suppose alternatively we could have the component explicitly emit an event when it finishes submitting, but I don't know if that's really worth doing just for the tests.